src/util/ThreadSafeQueue.h

```diff
@@ -278,10 +278,17 @@ ad_utility::InputRangeTypeErased<typename Queue::value_type> queueManager(
   AD_CONTRACT_CHECK(numThreads > 0u);
   using V = typename Queue::value_type;
   struct QueueGenerator : public ad_utility::InputRangeFromGet<V> {
-    std::vector<ad_utility::JThread> threads_;
+    // The order of these members is important, see the comment for the
+    // destructor.
     Queue queue_;
+    std::vector<ad_utility::JThread> threads_;
     std::atomic<int64_t> numUnfinishedThreads_;
 
+    // We first `finish` the queue. This allows the `threads_` to join in their
+    // destructor. Only then we can destroy the `queue_` (it is declared before
+    // the `threads_`, because no one is accessing it anymore.
+    ~QueueGenerator() { queue_.finish(); }
```

Comment:
Thanks @joka921, my point still stands that the call to `finish()` in the `ThreadSafeQueue` destructor doesn't do anything and should be removed because it could encourage wrong patterns like this one.